cmd/kube-proxy/app/server.go

```diff
@@ -621,6 +621,46 @@ func serveMetrics(bindAddress, proxyMode string, enableProfiling bool, errCh cha
 		return
 	}
 
+	if _, ipNet, err := net.ParseCIDR(bindAddress); err == nil {
+		// bindAddress is a subnet, it will get the IP that meets the subnet
+		if ifaces, err := net.Interfaces(); err == nil {
+			for _, i := range ifaces {
+				if addrs, err := i.Addrs(); err == nil {
```

Comment:
Pretty sure there's already a function in `k8s.io/apimachinery/pkg/util/net` that does this

We should have the same behavior for `--bind-address`, `--healthz-bind-address`, and `--metrics-bind-address`. Also, the command-line docs earlier in this file need to be updated, and likewise the config file docs in `pkg/proxy/apis/config/types.go`.

But probably there also needs to be a KEP? Or at least some discussion somewhere...